pwnlib/tubes/process.py

```diff
@@ -944,7 +944,10 @@ def maps(self):
         # addr = address (alias) = start (alias)
 
         from pwnlib.util.proc import memory_maps
-        raw_maps = memory_maps(self.pid)
+        raw_maps = self.poll() is None and memory_maps(self.pid)
+
+        if not raw_maps:
+            raise RuntimeError("Could not read maps, process %s has finished" % self.pid)
```

Comment:
Hm, changing the behavior here to throw instead of returning the `ELF.maps` mapping seems like a big change in behavior that could be talked about. Why do you think this is better?

Can we reduce this PR to just the `is not None` goof up fix instead of doing multiple changes at once please? That is an obvious typo :D